src/System.Private.ServiceModel/tests/Common/Scenarios/ServiceInterfaces.cs

```diff
@@ -39,6 +39,16 @@ public interface IWcfService
     void TestFaultInt(int faultCode);
 
     [OperationContract]
+    [FaultContract(typeof(FaultDetail), Action = "http://tempuri.org/IWcfService/TestFaultFaultDetailFault", Name = "FaultDetail", Namespace = "http://www.contoso.com/wcfnamespace")]
+    [FaultContract(typeof(FaultDetail2), Action = "http://tempuri.org/IWcfService/TestFaultFaultDetailFault2", Name = "FaultDetail2", Namespace = "http://www.contoso.com/wcfnamespace")]
+    void TestFaults(string faultMsg, bool flag);
+
+    [OperationContract]
+    [FaultContract(typeof(FaultDetail), Action = "http://tempuri.org/IWcfService/TestFaultFaultDetailFault", Name = "FaultDetail", Namespace = "http://www.contoso.com/wcfnamespace")]
+    [ServiceKnownType(typeof(FaultDetail2))]
```

Comment:
The known type, FaultDetail2, is not used in FaultContract; instead, it's used as the input of the operation. The name, FaultDetail2, doesn't accurately reflect how we use the type.